### PR TITLE
support any number of self-setup validators on a single host

### PR DESCRIPTION
**Pull request by rob-solana**

### CODE REVIEW

**Comment by mvines** [comment]
How about throwing a second validator into https://github.com/solana-labs/solana/blob/master/ci/localnet-sanity.sh as a part of this PR (or as a follow up)

File: multinode-demo/setup.sh [hunk 1]
```diff
@@ -77,12 +77,13 @@ mint_path="$SOLANA_CONFIG_PRIVATE_DIR"/mint.json
 
 set -e
 
-echo "Cleaning $SOLANA_CONFIG_DIR"
-rm -rvf "$SOLANA_CONFIG_DIR"
-mkdir -p "$SOLANA_CONFIG_DIR"
+for i in "$SOLANA_CONFIG_DIR" "$SOLANA_CONFIG_PRIVATE_DIR" "$SOLANA_CONFIG_VALIDATOR_DIR"
```
Comment (mvines) [inline multinode-demo/setup.sh]: nit: I prefer this style (fwiw):
```bash
for i in "$SOLANA_CONFIG_DIR" "$SOLANA_CONFIG_PRIVATE_DIR" "$SOLANA_CONFIG_VALIDATOR_DIR"; do
```

File: multinode-demo/validator.sh [hunk 2]
```diff
@@ -16,6 +16,13 @@ if [[ "$1" = "-h" || -n "$3" ]]; then
   usage
 fi
 
+if [[ "$1" == "-x" ]]; then
```
Comment (mvines) [inline multinode-demo/validator.sh]: double quotes around `$1` not needed right?

File: multinode-demo/validator.sh [hunk 3]
```diff
@@ -55,19 +58,30 @@ else
   program="$solana_fullnode"
 fi
 
+if (( ! self_setup)); then
```
Comment (mvines) [inline multinode-demo/validator.sh]: nit: inconsistent whitespacing

File: multinode-demo/validator.sh (hunk 2 above)
Comment (mvines) [reply multinode-demo/validator.sh]: Let's mention `-x` in `usage` for some self-documentation please

File: multinode-demo/validator.sh [hunk 4]
```diff
@@ -55,19 +58,30 @@ else
...
+  $solana_keygen -o "$validator_id_path"
 
-[[ -f "$SOLANA_CONFIG_DIR"/validator.json ]] || {
-  echo "$SOLANA_CONFIG_DIR/validator.json not found, create it by running:"
-  echo
-  echo "  ${here}/setup.sh"
-  exit 1
-}
+  validator_json_path="$SOLANA_CONFIG_VALIDATOR_DIR"/validator-x$$.json
+  $solana_fullnode_config --keypair="$validator_id_path" -l -b $((9000+($$%1000))) > "$validator_json_path"
```
Comment (mvines) [inline multinode-demo/validator.sh]: nit: My brain reads `$((9000 + ($$ % 1000)))` much easier than `$((9000+($$%1000)))`

File: multinode-demo/validator.sh (hunk 2 above)
Comment (rob-solana) [reply multinode-demo/validator.sh]: I was adhering to the style of the rest of the script, which is full of double quoting inside [[

**Pull request merged**